src/PowerShellEditorServices.Engine/Services/Workspace/ConfigurationService.cs

```diff
@@ -0,0 +1,13 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.PowerShell.EditorServices
+{
+    public class ConfigurationService
+    {
+        // This probably needs some sort of lock... or maybe LanguageServerSettings needs it.
```

Comment:
Perhaps the value should be a readonly copy of the settings value when this getter is called. There's still a data race in that case, but not a huge deal. If we lock and the getter wins, it's just held up an update to get an obsolete value. Ideally getting it would be a cancellable/retryable task (but then risks thread starvation). Anyway, readonly values always seem like the easiest way to be threadsafe to me